### PR TITLE
Plumb empty test_list when using --affected

### PR DESCRIPTION
**Pull request by lukebjerring**
This changes the filtering setup for `TestLoader` to treat an empty `include` list differently than a null one. Also changes the plumbing for wptcommandline to forward an empty `test_list` when `--affected` was specified, and to not pop `--affected` from the args to be able to make this check later.
The error-code of the run is also affected, with a value of `0` for no executed tests.
Executing without specifying a whitelist of test paths has the same behaviour as before (runs the full test suite as per the manifest.)
The effect on the output is that we still produce a report, which has `results: []`, rather than no report file, for the case that `--affected` produces no test results.

Fixes https://github.com/web-platform-tests/wpt/issues/14471

### CODE REVIEW

**Comment by Hexcles** [comment]
Sorry I didn't see this one before sending out my #14649.

This fix is a bit problematic because of a few reasons:
1. We are passing an extra arg via `kwargs` to `wptrunner`, which isn't defined in `wptrunner`'s argparser. It'd be very difficult later on to figure out where this field comes from and what it means. Adding custom fields to the dict populated by `argparser` is surprising.
2. I don't like relying on `[]` vs `None`. Yes, they are different and can be used this way, but I usually find it hard to maintain and error-prone in the long run.

Basically, I'd like to make things more explicit as in #14649 .
